packages/server/src/fhir/repo.ts

```diff
@@ -1019,6 +1019,13 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
     try {
       const resource = await this.readResourceImpl(resourceType, id);
 
+      if (patch.find((operation) => operation.path === '/resourceType')) {
+        throw new OperationOutcomeError(badRequest('Incorrect Resource Type'));
+      }
+      if (patch.find((operation) => operation.path === '/id')) {
+        throw new OperationOutcomeError(badRequest('Incorrect Id'));
+      }
```

Comment:
I'd prefer for these checks to live in `updateResourceImpl()`, which several different `Repository` methods call into, so that this issue can be prevented in depth for the future.  @gusIreland If you don't mind making that change I'd appreciate it, but I'm also happy to handle it if you're short on time — let me know either way!

Reply:
That makes sense! I'll update it and ping you when it's ready